### DISCUSSION TITLE
Root middleware for Query type.

### DISCUSSION THREAD

**Discussion by valerii15298** [MichalLytek/type-graphql#1280] [Q&A]
I have query like this:
```graphql
query {
  projects {
    id
  }
}
```
 What I need is middleware which will be invoked only on query type.
So that using `graphql-fields` plugin or `graphql-fields-list` I can get list of all root fields(including field `projects`) from current query.
Is it possible? And how to do it? Maybe some different, better way to achieve it?
Thanks in advance!!!

Global middleware would be a solution but unfortunately is not executed on Query type(I tried) so I cannot use it.

**Comment by MichalLytek** [comment]
> Global middleware would be a solution but unfortunately is not executed on Query type(I tried) so I cannot use it.

It should be executed, middlewares are executed for each resolver, both query, mutation and field resolvers.
